lxd-agent/metrics.go

```diff
@@ -0,0 +1,371 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/lxc/lxd/lxd/response"
+	"github.com/lxc/lxd/shared/api"
+	"github.com/lxc/lxd/shared/filesystem"
+	"github.com/pkg/errors"
+)
+
+var defMountPointsExcluded = regexp.MustCompile("^/(dev|proc|sys|var/lib/docker/.+)($|/)")
```

Comment:
Why is var/lib/docker excluded? Feels like a comment would be useful here for the future.

Reply:
Could this be replace with a prefix match on a slice of strings, using `shared.StringHasPrefix()`, maybe combined with stripping any trailing slashes first?

Reply:
We could do it differently. I just copied the code from the node_exporter.

Reply:
OK, well up to you, I tend to err towards avoiding regexes if possible.

Reply:
I stuck with the regex for the mountpoints.